src/wp-includes/theme.php

```diff
@@ -832,6 +853,13 @@ function switch_theme( $stylesheet ) {
 
 	update_option( 'theme_switched', $old_theme->get_stylesheet() );
 
+	/*
+	 * Reset globals to force refresh the next time these directories are
+	 * accessed via `get_stylesheet_directory()` / `get_template_directory()`.
+	 */
+	$wp_stylesheet_path = null;
+	$wp_template_path   = null;
+
```

Comment:
The creation of `wp_stylesheet_path` and `wp_template_path` global will lead to a slim chance of people using the global directly rather that via the function (as we can't force them to stick to functions only using some notice or error message)

Incase a sites depending extensively on global  will get error when they `switch_theme`, as both the path globals are going to be null. I would prefer us updating the global immediately after `switch_theme`

Reply:
@kt-12 IMO that's not something to be concerned of. WordPress uses lots of globals, and generally those globals should never be used directly, except for a few exceptions such as `$wpdb`.

We could maybe rename them to use an underscore prefix to clarify via code, i.e. `$_wp_*`?

Reply:
I agree with @felixarntz's thought here. If people are getting results from the function, they might not use global variables, but we can't restrict them from using globals.

Reply:
@felixarntz I don't think we should rename itto `$_wp_*`. I was thinking if we could reinitialise global immediately after we `switch_theme` the value obtained from usage of global will also be true at all given points. 
Anyway, it's a very rare scenario(unless proven otherwise in future) that someone would opt for global instead of the function so we can leave it the way it is.